src/Orckestra.Composer.Cart/Providers/Order/EditingOrderProvider.cs

```diff
@@ -53,6 +54,39 @@ public virtual async Task<bool> CanEdit(Overture.ServiceModel.Orders.Order order
             return isOrderEditable;
         }
 
+        public virtual async Task<bool> IsOrderCancelable(Overture.ServiceModel.Orders.Order order)
+        {
+            if (order.OrderStatus.Equals("Canceled", StringComparison.InvariantCultureIgnoreCase)) return false;
+
+            var orderFulfillmentState = await OrderRepository.GetOrderFulfillmentStateAsync(new GetOrderFulfillmentStateParam
+            {
+                OrderId = order.Id,
+                ScopeId = order.ScopeId
+            }).ConfigureAwait(false);
+
+            if (orderFulfillmentState.IsCancelable 
```

Comment:
also add additional check: fulfilment state contains something like allowednextstates, check if this contains "Canceled"